metrics/collector_test.go

```diff
@@ -125,6 +125,91 @@ func testMachineStatus(t *testing.T) {
 	}
 }
 
+func MachineStatusWhenMachineDeleted(t *testing.T) {
+	testCases := []machineStatusTestCase{
```

Comment:
Why do you write this as a table test?
This test cannot be generalized, so please write it normally.